### ISSUE TITLE
Display the subComponents API inside the Controls addons (Canvas)

### ISSUE THREAD

**Issue by alex83130** [storybookjs/storybook#13379]
**Is your feature request related to a problem? Please describe**
In the same logic we have access to the main component's API inside the Canva's addons panel with `Controls`.
Is there a way to include the subComponent's API as well ? (As it's display inside the Docs)

**Additional context**

The idea will be to have that UI which include the subComponent notion :
<img width="1044" alt="Screen Shot 2020-12-04 at 1 47 30 PM" src="https://user-images.githubusercontent.com/15250423/101202517-60cbc400-3637-11eb-9f62-bc4e23e5a5ac.png">

Instead of only :
<img width="641" alt="Screen Shot 2020-12-04 at 1 46 14 PM" src="https://user-images.githubusercontent.com/15250423/101202312-19453800-3637-11eb-9071-2b7864d86866.png">

When there are subComponent

**Comment by ghengeveld** [comment]
If I understand correctly, you have a component with static properties, which are components themselves? E.g.:

```jsx
// MyComponent.js
const MyComponent = (props) => <div>Hello</div>

// SubComponent as static property
MyComponent.SubComponent = (props) => <div>World</div>

export default MyComponent
```

Naturally, you would have a `MyComponent.stories.js`, and you could define the stories for `MyComponent.SubComponent` in the same file. However that means you won't have the flexibility to change the story hierarchy that way. Therefore I recommend you also create a `MyComponent.SubComponent.stories.js` (for each sub component) and set the proper `title`. That would give you something like this:

<img width="203" alt="Screenshot 2021-03-18 at 10 38 39" src="https://user-images.githubusercontent.com/321738/111605030-3e207e00-87d6-11eb-8802-9b0b6499fb0a.png">

This is not at all like your proposal to have a separate tab for each subcomponent, but it fits the current model. I don't see a real benefit to multiple canvas tabs to be honest, at least not for such a specific use case.

Please correct me if I misunderstood or if you have a specific workflow where my suggestion doesn't work.

**Comment by shilman** [comment]
This issue is about how subcomponents are displayed in the controls addon, not about the sidebar @ghengeveld

**Comment by alex83130** [comment]
Exactly, within the controls addons tab,  the feature would be to be able to "play" with a SubCompoment in the same way we are for the moment able to play with the MainComponent.

To give you a simple example and a bit more context : 
We have within our storybook a Modal component which use the Compound component pattern, this one is pretty simlare to that code below : 

```
<Modal type={}>
    <Modal.Title format={}>A Title</Modal.Title>
    <Modal.Content>Lorem ipsum dolor sit amet, consectetur adipiscing.</Modal.Title>
    <Modal.Button disabled={}>Valider</Modal.Title>
</Modal>
```

For the moment the other devs are able to "play" with `Modal` (change the `type`).
But cannot directly "play" with `Modal.Title` (change the `format`) or with `Modal.Button` (make the button disabled or not).

The solution we have for the time being is to create many sub-stories in order to reproduce these "sub-cases", but it could be great if everything is part of the controls addons mindset

**Comment by sami616** [comment]
Yes, there is a separate issue (#10979) here. Would love to see this make it into a future release. 

Perhaps something like this might be a nice API.

![94360539-e5cdb680-00a5-11eb-8058-0f7f2b4db15a](https://user-images.githubusercontent.com/21065581/112499453-63396180-8d7f-11eb-9b53-74a9548019aa.png)
